Realm/RLMSyncUtil.h

```diff
@@ -29,19 +29,46 @@ extern NSString *const kRLMSyncPathOfRealmBackupCopyKey;
 /// A user info key for use with `RLMSyncErrorClientResetError`.
 extern NSString *const kRLMSyncInitiateClientResetBlockKey;
 
+/**
+ A user info key for use with `RLMSyncErrorAuthError`.
+
+ The value associated with this key is the `RLMSyncAuthError` integer error code which describes the specific
+ authentication or permissions-related error situation.
+ */
+extern NSString *const kRLMSyncUnderlyingAuthErrorCodeKey;
+
 /// The error domain string for all SDK errors related to synchronization functionality.
 extern NSString *const RLMSyncErrorDomain;
 
 /// An error which is related to authentication to a Realm Object Server.
 typedef RLM_ERROR_ENUM(NSInteger, RLMSyncAuthError, RLMSyncErrorDomain) {
-    /// An error that indicates that the provided credentials are invalid.
-    RLMSyncAuthErrorInvalidCredential   = 611,
+    /**
+     An error that indicates that the provided credentials are invalid. This error may also
+     indicate an attempt to log in an existing user that does not actually exist, or to create
+     a new user when an existing user already has the given username.
+    */
+    RLMSyncAuthErrorInvalidCredential               = 611,
+
+    /// Do not use.
```

Comment:
not quite do not use, since these may be reported if the current user is an admin